components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/codegen/CodeGenerator.java

```diff
@@ -86,6 +88,47 @@ public void generate(String projectName, List<ExtendedAPI> apis, boolean overwri
 
     }
 
+    /**
+     * Generates ballerina source for provided Open APIDetailedDTO Definition in {@code definitionPath}.
+     * Generated source will be written to a ballerina package at {@code outPath}
+     * <p>Method can be user for generating Ballerina mock services and clients</p>
+     *
+     * @param projectName   name of the project being set up
+     * @param apiDef        api definition string
+     * @param endpointDef   endpoint definition string
+     * @param overwrite     whether existing files overwrite or not
+     * @throws IOException                  when file operations fail
+     * @throws BallerinaServiceGenException when code generator fails
+     */
+    public void generate(String projectName, String apiDef, String endpointDef, boolean overwrite)
+            throws IOException, BallerinaServiceGenException {
+        BallerinaService definitionContext;
+        SwaggerParser parser;
+        Swagger swagger;
+        String projectSrcPath = GatewayCmdUtils.getProjectSrcDirectoryPath(projectName);
+        List<GenSrcFile> genFiles = new ArrayList<>();
+
+        parser = new SwaggerParser();
+        swagger = parser.parse(apiDef);
+        ExtendedAPI api = new ExtendedAPI();
+        api.setName(swagger.getInfo().getTitle());
```

Comment:
do we have a builder pattern for extendedAPI class?

Reply:
in the normal scenario, the extended api class is populated via an object mapper. But in this case we will have to use the builder pattern because, the ExtendedAPI class only have the default constructor.